src/config.py

```diff
@@ -1,10 +1,66 @@
 import os
+import psycopg2
+from sqlalchemy import create_engine, Engine
 from dotenv import load_dotenv
+from urllib.parse import urlparse
+from psycopg2.extensions import connection as Psycopg2Connection
+from src.helper_functions import get_logger
 
 load_dotenv()
 ETHEREUM_NODE_URL = os.getenv("ETHEREUM_NODE_URL")
 GNOSIS_NODE_URL = os.getenv("GNOSIS_NODE_URL")
 
 CHAIN_RPC_ENDPOINTS = {"Ethereum": ETHEREUM_NODE_URL, "Gnosis": GNOSIS_NODE_URL}
 
-CHAIN_SLEEP_TIMES = {"Ethereum": 60, "Gnosis": 120}
+
+# function for safe conversion to float (prevents None -> float conversion issues raised by mypy)
+def get_env_float(var_name: str) -> float:
+    """Retrieve environment variable and convert to float. Raise an error if not set."""
+    value = os.getenv(var_name)
+    if value is None:
+        raise ValueError(f"Environment variable {var_name} is not set.")
+    try:
+        return float(value)
+    except ValueError:
+        raise ValueError(f"Environment variable {var_name} must be a float.")
```

Comment:
Do we want to auto convert ints to floats?

Reply:
Thing is, getenv() default is string, so conversion seemed needed, but now I've changed it to int instead of float since its more suitable for sleep time.